File: shortest path/floydWarshall.java
import java.util.Arrays;
import java.util.Scanner;

public class floydWarshall {

    public static final Scanner sc = new Scanner(System.in);
    public static final int INF = (int) 1e9;
    public static int[][] graph = new int[501][501];
    public static int n,m;

    public static void main(String[] args) {
        n=sc.nextInt();
        m=sc.nextInt();

        for(int i=0; i<501;i++){
            Arrays.fill(graph[i],INF);
        }

        for(int i=0;i<501;i++){
            graph[i][i]=0;
        }

        for(int i=0; i<m;i++){
            int a= sc.nextInt();
            int b= sc.nextInt();
            int c= sc.nextInt();
            graph[a][b]=c;
        }

        for(int i=1; i<=n;i++){
            for(int j=1; j<=n;j++){
                for(int k=1; k<=n;k++){
                    graph[j][k]=Math.min(graph[j][k], graph[j][i]+graph[i][k]);
                }
            }
        }

        for(int i=1; i<=n;i++){
            for(int j=1; j<=n;j++){
                if(graph[i][j]==INF){
                    System.out.print("INF ");
                }else{
                    System.out.print(graph[i][j]+" ");
                }
            }
            System.out.println();
        }



    }

}

File: shortest path/floydWarshall.py
INF = int(1e9)

n = int(input())
m = int(input())

graph = [[INF] * (n + 1) for _ in range(n + 1)]

for a in range(1, n + 1):
    # for b in range(1, n+1):
    #     if a == b:
    graph[a][a] = 0

for _ in range(m):
    a, b, c = map(int, input().split())
    graph[a][b] = c

for i in range(1, n + 1):
    for j in range(1, n + 1):
        for k in range(1, n + 1):
            graph[j][k] = min(graph[j][k], graph[j][i] + graph[i][k])


for a in range(1,n+1):
    for b in range(1,n+1):
        if graph[a][b] == INF:
            print("INF", end=" ")
        else:
            print(graph[a][b], end=" ")
    print()
